### DISCUSSION TITLE
Question for querying with an array of ids

### DISCUSSION THREAD

**Discussion by ashiva** [supabase/supabase#2576] [Questions] [answered]
Hi,

New to Supabase and dev in general, and have a question. 

I am currently saving an array of ids i.e. `[1,4,6,23]` in one column on a 'jobs' table  as an association to a 'skills' table (one to many) - and I was wondering what's the best way to use the Supabase to pull the associated 'skills' records for a given job by passing in the array? 

My current query looks like:

```
.from("jobs")
    .select(`
    *,
    categories (
      name
    ),
    profiles (
      company, name, website
    )`
    )
    .filter("id", "eq", id)
    .single();
```

**Comment by steve-chavez** [comment, marked as answer]
Hey @ashiva,

Currently it's not possible to query a foreign table through an array - this is because PostgreSQL doesn't support foreign keys arrays. For a workaround, see this previous discussion: https://github.com/supabase/supabase/discussions/535#discussioncomment-285696

**Comment by ashiva** [comment]
@steve-chavez

Thanks, I got it to work with two different calls, but now I sometimes run into an issue where the 2nd call is dependent on the data from the first, anyway to "chain" them so it doesn't return undefined in the 2nd call?:
```
    const { data } = await supabase
      .from("jobs")
      .select(
        `*,
        categories (
          name
        ),
        community (
          name
        )`
      )
      .filter("id", "eq", id)
      .single();

    const { data: skills } = await supabase
      .from("skills")
      .select(`title`)
      .in("id", data.skills_id || []);
```

**Reply by steve-chavez** [reply]
> where the 2nd call is dependent on the data from the first, anyway to "chain" them so it doesn't return undefined in the 2nd call?

Yeah, for this you need referential integrity, which is what foreign keys give you. I'd suggest using the approach I mentioned in the other [answer](https://github.com/supabase/supabase/discussions/2576#discussioncomment-1058622).
